pilot/pkg/bootstrap/server.go

```diff
@@ -296,8 +311,22 @@ func (s *Server) initKubeClient(args *PilotArgs) error {
 	return nil
 }
 
+// A single container cannot have two readiness probes. Piggyback the https server
+// onto the http server readiness.
+func (s *Server) ready(w http.ResponseWriter, req *http.Request) {
+	if status := s.httpsServerReady(); status != http.StatusOK {
+		log.Warnf("https webhook server not ready: %v", status)
+		w.WriteHeader(status)
+		return
+	}
+
+	w.WriteHeader(http.StatusOK)
```

Comment:
nit: this was a bit confusing to me, since it looks like its "only" checking the https part. The key here is that the fact that we have started listening on this means that we have already initialized everything else (granted, this is not a great check). May be worth commenting that?